types/massive/index.d.ts

```diff
@@ -42,194 +49,338 @@ declare namespace massive {
+    }
+
+    interface DecomposeOptions {
+        pk: string;
+        columns?: string[] | AnyObject<string>;
+        [foreignTable: string]: DecomposeOptions | any;
     }
 
     interface SearchDefinition {
-        /** List of the fields to search. */
         fields: string[];
-
-        /** Search term. */
         term: string;
-    }
-
-    interface QueryOptions {
-        /** This is a query against a document table. */
-        document?: boolean;
-        /** True to return a single result object instead of an array of results. */
-        single?: boolean;
-        /** True to return a stream instead of a resultset. */
-        stream?: boolean;
+        where: AnyObject;
     }
 
     interface SearchCriteria {
         fields: string[];
         term: string;
     }
 
-    class Table extends Queryable {
-        /** Delete a record or records. */
-        destroy(criteria: object, options?: GenericQueryOptions): Promise<void>;
-
-        /** Attempts to assemble primary key criteria for a record object representing a row in this table. The criteria must include the full primary key, and must not invoke any operations. */
-        getPkCriteria(record: object): object;
-
-        /** Insert a record or records into the table. */
-        insert(data: object | any[], options?: GenericQueryOptions): Promise<any>;
-
-        /**
-         * Update a document, adding new information and changing existing information.
-         * This function can be used with any JSON field, not just document tables; however, only document tables can use criteria objects which directly reference document fields.
-         * If calling modify with a criteria object for a non-document table, the criteria will be tested against the entire row (as opposed to the document body as it is for document tables).
-         * To test elements of the JSON field in a non-document table with a criteria object, use a JSON path string.
-         */
-        modify(criteria: object | UUID, changes: object, field?: string): Promise<any>;
-
-        /**
-         * Performs an upsert.
-         * If the record does not include a value for the primary key column, it will be inserted and the persisted record (including primary key) returned;
-         * if it does, the row will be updated and the modified record returned.
-         */
-        save(record: object, options?: GenericQueryOptions): Promise<any>;
-
-        /**
-         * Save a document to the database. This function replaces the entire document body.
-         */
-        saveDoc(record: object): Promise<any>;
-
-        /**
-         * Update a record.
-         * May be invoked with a complete record (including primary key), or with a criteria object and a map of fields to new values.
-         * Multi-row updates are only possible through the latter usage.
-         */
-        update(criteria: object, fields: object, options?: GenericQueryOptions): void;
-    }
+    class Table extends Writable {}
```

Comment:
I don't use TypeScript so this may be doing something I don't know about but there's no longer a `Table` module. A database table is represented as a `Writable` (which implicitly applies the `Readable` prototype).

Reply:
This was a holdover from the previous version. This line is basically a type alias for the `Writable` class—I don't find much value other than the name, so I'll probably just remove it.